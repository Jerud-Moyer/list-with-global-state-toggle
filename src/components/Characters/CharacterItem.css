.CharacterItem {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;


  figcaption {
    font-size: 25px;
   
  }
  
  

  img {
    height: 150px;
    width: 200px;
  }

}
