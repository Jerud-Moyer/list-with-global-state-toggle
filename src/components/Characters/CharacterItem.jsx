import React from 'react';
import PropTypes from 'prop-types';
import styles from './CharacterItem.css';

const CharacterItem = ({ name, imageUrl }) => {
  return (
    <figure className={styles.CharacterItem}>
      <img src= {imageUrl} alt={name} />
      <figcaption>{name}</figcaption>
    </figure>
  );
};

CharacterItem.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired
};

export default CharacterItem;
