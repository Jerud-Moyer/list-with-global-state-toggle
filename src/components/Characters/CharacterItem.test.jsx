import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterItem from './CharacterItem';

describe('CharacterItem component', () => {
  afterEach(() => cleanup());

  it('renders Character', () => {
    const { asFragment } = render(<CharacterItem
      imageUrl="Test Image"
      name="test Name"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
