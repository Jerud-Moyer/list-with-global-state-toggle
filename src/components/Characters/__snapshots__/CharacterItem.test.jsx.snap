// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem component renders Character 1`] = `
<DocumentFragment>
  <figure
    class="CharacterItem"
  >
    <img
      alt="test Name"
      src="Test Image"
    />
    <figcaption>
      test Name
    </figcaption>
  </figure>
</DocumentFragment>
`;
